config/routes.rb

```diff
@@ -597,20 +598,20 @@
   resources :similar_topics
 
   get "topics/feature_stats"
-  get "topics/created-by/:username" => "list#topics_by", as: "topics_by", constraints: { username: RouteFormat.username }
-  get "topics/private-messages/:username" => "list#private_messages", as: "topics_private_messages", constraints: { username: RouteFormat.username }
-  get "topics/private-messages-sent/:username" => "list#private_messages_sent", as: "topics_private_messages_sent", constraints: { username: RouteFormat.username }
-  get "topics/private-messages-archive/:username" => "list#private_messages_archive", as: "topics_private_messages_archive", constraints: { username: RouteFormat.username }
-  get "topics/private-messages-unread/:username" => "list#private_messages_unread", as: "topics_private_messages_unread", constraints: { username: RouteFormat.username }
-  get "topics/private-messages-group/:username/:group_name.json" => "list#private_messages_group", as: "topics_private_messages_group", constraints: {
-    username: RouteFormat.username,
-    group_name: RouteFormat.username
-  }
-
-  get "topics/private-messages-group/:username/:group_name/archive.json" => "list#private_messages_group_archive", as: "topics_private_messages_group_archive", constraints: {
-    username: RouteFormat.username,
-    group_name: RouteFormat.username
-  }
+
+  scope "/topics", username: RouteFormat.username do
+    get "created-by/:username" => "list#topics_by", as: "topics_by"
+    get "private-messages/:username" => "list#private_messages", as: "topics_private_messages"
+    get "private-messages-sent/:username" => "list#private_messages_sent", as: "topics_private_messages_sent"
+    get "private-messages-archive/:username" => "list#private_messages_archive", as: "topics_private_messages_archive"
+    get "private-messages-unread/:username" => "list#private_messages_unread", as: "topics_private_messages_unread"
+    get "private-messages-tag/:username/:tag_id.json" => "list#private_messages_tag", as: "topics_private_messages_tag", constraints: StaffConstraint.new
```

Comment:
We have support for filtering [multiple tags in public topics](https://meta.discourse.org/t/tag-intersections-page/47952) so we should probably do the same for private messages. Might in the next iteration?